beacon_node/eth2_libp2p/src/peer_manager/peer_info.rs

```diff
@@ -309,8 +326,19 @@ impl PeerConnectionStatus {
 
     pub fn connections(&self) -> (u8, u8) {
         match self {
-            Connected { n_in, n_out } => (*n_in, *n_out),
+            Connected { n_in, n_out, .. } => (*n_in, *n_out),
             _ => (0, 0),
         }
     }
+
+    /// Returns true iff the peer is connected and its first connection was outgoing.
+    pub fn is_outgoing(&self) -> bool {
```

Comment:
Might be nice adding `is_incoming` as well? Because it's not quite equal to `!is_outgoing`